Library/Homebrew/extend/ENV/super.rb

```diff
@@ -50,7 +50,7 @@ def setup_build_environment(formula = nil)
     self["PKG_CONFIG_PATH"] = determine_pkg_config_path
     self["PKG_CONFIG_LIBDIR"] = determine_pkg_config_libdir
     self["HOMEBREW_CCCFG"] = determine_cccfg
-    self["HOMEBREW_OPTIMIZATION_LEVEL"] = "Os"
+    self["HOMEBREW_OPTIMIZATION_LEVEL"] = "O0"
```

Comment:
You should be able to set this inside your formula with `ENV.O0`. I think that's right anyway. The maintainer I would normally ask about compiler optimisation issues is away at the moment :smile:.